File: src/cxx/cpp/qnetwork/cxx_qnetwork.cpp
#include "aurora-rs-mcp/src/cxx/cpp/qnetwork/cxx_qnetwork.h"
#include "aurora-rs-mcp/src/cxx/rust/cxx_qnetwork.rs.h"

namespace com
{
  namespace keygenqt
  {
    namespace aurora_rs_mcp
    {
        std::unique_ptr<CxxQnetwork> new_cxx_qnetwork()
        {
            return std::make_unique<CxxQnetwork>();
        }

        CxxQnetwork::CxxQnetwork() {}

        bool CxxQnetwork::is_online() const
        {
            return m_manager.isOnline();
        }
    } // namespace aurora_rs_mcp
  } // namespace keygenqt
} // namespace com

#include "cxx_qnetwork.moc"

File: src/cxx/rust/cxx_qnetwork.rs
#[cxx::bridge(namespace = "com::keygenqt::aurora_rs_mcp")]
mod ffi {
    unsafe extern "C++" {
        include!("aurora-rs-mcp/src/cxx/cpp/qnetwork/cxx_qnetwork.h");

        type CxxQnetwork;

        fn new_cxx_qnetwork() -> UniquePtr<CxxQnetwork>;
        fn is_online(&self) -> bool;
    }
}

/// Example execute
pub fn is_online() -> bool {
    ffi::new_cxx_qnetwork().is_online()
}
